watcher/watcher.go

```diff
@@ -240,9 +256,113 @@ func (w *Watcher) reportRoomsStatusesRoutine() {
 func (w *Watcher) watchRooms() error {
 	w.RemoveDeadRooms()
 	w.AutoScale()
+	w.AddUtilizationMetricsToRedis()
 	return nil
 }
 
+// AddUtilizationMetricsToRedis store the pods usage metrics (cpu and mem) in a redis sorted set
+func (w *Watcher) AddUtilizationMetricsToRedis() {
+	logger := w.Logger.WithFields(logrus.Fields{
+		"executionID": uuid.NewV4().String(),
+		"operation":   "addUtilizationMetricsToRedis",
+		"scheduler":   w.SchedulerName,
+	})
+	logger.Info("starting to add utilization metrics to redis")
+
+	scheduler, _, _, err := controller.GetSchedulerScalingInfo(
+		logger,
+		w.MetricsReporter,
+		w.DB,
+		w.RedisClient.Client,
+		w.SchedulerName,
+	)
+
+	if err != nil {
+		if strings.Contains(err.Error(), "not found") {
+			w.Run = false
```

Comment:
why should we stop the watcher if a scheduler is not found?

Reply:
Actually I've just copied the behaviour from `AutoScale()`. It will force the watcher to restart if the scheduler exists

Reply:
Ok, this makes sense. I made a confusion between the watcher and the worker.